src/compiler/checker/lines_12000_16000.rs

```diff
@@ -368,6 +368,17 @@ impl TypeChecker {
         unimplemented!()
     }
 
+    pub(super) fn get_type_id(&self, type_: Rc<Type>) -> TypeId {
+        type_.id()
+    }
+
+    pub(super) fn contains_type(&self, types: &[Rc<Type>], type_: Rc<Type>) -> bool {
+        /*binary_search(...)*/
```

Comment:
Is the list (eventually) sorted in some way that makes a binary search possible?

Reply:
Ya I guess it's not codified at the type level (eg by using `SortedArray`) but it does appear that the `.types` field on union/intersection types will always be sorted by type ID (and I'm mimicking that now in `.add_type_to_union()`)

So updated here to use `binary_search_copy_key()` now that that exists